content/blog/2022/2022-05-22-connect-to-aws-rds.md

```diff
@@ -0,0 +1,120 @@
+
+With our database created, we will next set up a jump host and populate inbound rules in the security groups in the following sections.
+
+## Creating an EC2 Instance as the Jump Host 
+A jump host is also called a bastion host/server whose sole purpose is to provide access to resources in a private network from an external network like the internet. A rough representation of this architecture is shown below:
+
+{{% image alt="Jump host" src="images/posts/aws-rds-connect/jump-host.png" %}}
+
+Here we are using an EC2 instance in a public subnet as our jump host for connecting to an RDS database in a private subnet.
+
+Let us create the EC2 instance from the AWS Management Console in a public subnet in the same VPC where we had created our RDS database in the previous section.:
+
+{{% image alt="Create EC2 bastion" src="images/posts/aws-rds-connect/create-ec2.png" %}}
+
+We have created our instance in the free tier with an SSH key pair to access the instance with SSH in the later sections. The private key of the SSH key pair is downloaded and saved to our local workstation. 
+
+For creating the instance in the public subnet we have used the network settings as shown below:
+{{% image alt="Create EC2 bastion" src="images/posts/aws-rds-connect/ec2-network.png" %}}
+
+We will use this EC2 instance as our jump host on which we will set up an SSH tunnel for connecting to the RDS database in the next section.
+
+## Allow Traffic to the RDS Database from the Jump Host
+To enable connectivity to our RDS database, any security groups, network ACL, security rules, or third-party security software that exist on the RDS database must allow traffic from the EC2 instance used as the jump host. 
+
+In our example, the security group of our RDS database must allow access to port `3306` from the EC2 instance. To enable this, let us add an inbound rule to the security group `db-sg` to allow connections from the EC2 instance:
+
+{{% image alt="Create RDS Database" src="images/posts/aws-rds-connect/added-ingress.png" %}}
+
+This EC2 instance is also secured by a security group. A security group by default is associated with an outbound rule which has all outbound traffic enabled. This will allow the EC2 instance to make an outbound connection to the RDS database.
+
+## Connecting to the RDS Database
+We use a mechanism called SSH tunneling or port forwarding for connecting to the RDS database. SSH tunneling is a method of transporting arbitrary networking data over an encrypted SSH connection.
+
+The SSH client on our local workstation listens for connections on a configured port. When it receives a connection, it tunnels the connection to the SSH server which is our EC2 instance acting as the jump host. 
+
+The SSH server(EC2 instance) connects to a destination port, usually on a different machine than the SSH server. In our example, the destination port is the port of the RDS database. 
```

Comment:
We can mention, that due to the private key on the local machine, the tunnel is secured/encrypted and the owner of the private key is authenticated by the jump host. Because this is the goal of tunneling.